includes/mailpoet/class-sensei-mailpoet.php

```diff
@@ -0,0 +1,385 @@
+<?php
+/**
+ * File containing the class Sensei_MailPoet.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( ! class_exists( \MailPoet\API\API::class ) ) {
+	return;
+}
+
+/**
+ * MailPoet integration class.
+ *
+ * Handles the integration with the MailPoet plugin,
+ * creates a list for each course and group, adds enrolled students.
+ *
+ * @package Core
+ * @since $$next-version$$
+ */
+class Sensei_MailPoet {
+
+	/**
+	 * MailPoet API handle.
+	 *
+	 * @var object
+	 */
+	private $mailpoet_api;
+
+	/**
+	 * Instance of the current handler.
+	 *
+	 * @var Sensei_MailPoet
+	 */
+	private static $instance;
+
+	/**
+	 * A list of Sensei Courses and Groups.
+	 *
+	 * @var array
+	 */
+	private $sensei_lists;
+
+	/**
+	 * All lists on MailPoet.
+	 *
+	 * @var array
+	 */
+	private $mail_poet_lists;
+
+	/**
+	 * Constructor
+	 *
+	 * @since $$next-version$$
+	 * @param object $mailpoet_api MailPoet API handle.
+	 */
+	public function __construct( $mailpoet_api ) {
+		$this->mailpoet_api = $mailpoet_api;
+		if ( $this->mailpoet_api->isSetupComplete() ) {
+			add_action( 'init', array( $this, 'maybe_schedule_sync_job' ), 10 );
+
+			add_action( 'sensei_pro_student_groups_group_student_added', array( $this, 'add_student_subscriber' ), 10, 2 );
+			add_action( 'sensei_pro_student_groups_group_students_removed', array( $this, 'remove_student_subscribers' ), 10, 2 );
+
+			add_action( 'sensei_course_enrolment_status_changed', array( $this, 'maybe_add_student_course_subscriber' ), 10, 3 );
+			add_action( 'sensei_admin_enrol_user', array( $this, 'add_student_subscriber' ), 10, 2 );
+			add_action( 'sensei_manual_enrolment_learner_enrolled', array( $this, 'add_student_course_subscriber' ), 10, 2 );
+			add_action( 'sensei_manual_enrolment_learner_withdrawn', array( $this, 'remove_student_course_subscriber' ), 10, 2 );
+		}
```

Comment:
I am probably missing something here but I have the impression that the `sensei_course_enrolment_status_changed` is called whenever the enrolment changes. Why do we need the 3 hooks after it?

Reply:
Hmmm this sounds like we should verify this and make sure that `sensei_course_enrolment_status_changed` is called whenever enrolment changes. But we should handle it as part of another PR.